### PR TITLE
Fix links track with use_middle

### PR DESCRIPTION
**Pull request by Jeff1995**
Proposed bug fix for #422.

### CODE REVIEW

**Comment by lldelisle** [comment]
Thank you very much for the bug fix, indeed, I did not think about this case.
What is missing to this PR is:
- target to develop instead of master and rebase or merge develop
- add a test so we never reintroduce this bug
- add a release text to announce this bug fix
Tell me if you are too busy to do it, I can do it as well.

**Comment by Jeff1995** [comment]
Sure! I have rebased and targeted to develop, and modified a test file to include a situation that would have raised the error. Release text has also been added. Note that the modifed test file "test_wide.arcs" causes change to three test images. Could you check and see if that's appropriate?

**Comment by lldelisle** [comment]
The test looks good thank you. I am just wondering if the output of master_arcs_use_middle.png should not be changed to show like we see on master_links_squares_overlay.png...

**Comment by Jeff1995** [comment]
I guess this is just a corner case where the triangular and square visualizations look awkward, and should not have been used in the first place.

**Comment by lldelisle** [comment]
Yes, I think you are right, we never thought that end2 could be smaller than end 1....
I guess the expected behaviour would be to go from start1 to max(end1, end2)

**Comment by lldelisle** [comment]
What I was mentioning is what I draw in pink which is missing:
image
Also because we never thought about the case end2>end1

**Comment by Jeff1995** [comment]
Thanks! Now I understand 😀. I agree with these expected behaviors. I'll see what I can do and get back to you tomorrow.

**Comment by lldelisle** [comment]
Thank you very much. If you don't have time, I can do it.

**Comment by Jeff1995** [comment]
I think I've got it to work.

Screen Shot 2022-12-02 at 14 22 06

**Comment by lldelisle** [comment]
This is super great! Thank you so much.

File: docs/content/releases/3.7.1.rst [hunk 1]
```diff
@@ -5,4 +5,4 @@ Bugfix:
 ^^^^^^^
 
 - Works with gffutils version 0.11.0 where ValueError became gffutils.exceptions.EmptyInputError for empty gtf.
-
+- Fixed a small bug in links track with "use_middle" enabled, where midpoints could have different ordering than start points.
```
Comment (lldelisle) [inline docs/content/releases/3.7.1.rst:8]:
```suggestion
- Fixed a small bug in links track with "use_middle" enabled, where midpoints could have different ordering than start points (Thanks to @Jeff1995 ).
```

**Review by lldelisle — APPROVED** [review]

**Pull request merged**